core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -71,6 +72,13 @@ public HadoopFileIO(Configuration hadoopConf) {
 
   public HadoopFileIO(SerializableSupplier<Configuration> hadoopConf) {
     this.hadoopConf = hadoopConf;
+    Map<String, String> props = Maps.newHashMapWithExpectedSize(hadoopConf.get().size());
```

Comment:
These bugs were discovered during serialization test (don't remember in core module or flink module)

Reply:
can we extract this into a separate PR maybe (with separate tests)?

Reply:
https://github.com/apache/iceberg/pull/10926

will revert the change in this PR